src/neo4j_graphrag/retrievers/tools_retriever.py

```diff
@@ -0,0 +1,161 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+from typing import Any, List, Optional, Sequence
+
+import neo4j
+
+from neo4j_graphrag.llm.base import LLMInterface
+from neo4j_graphrag.retrievers.base import Retriever
+from neo4j_graphrag.types import RawSearchResult
+from neo4j_graphrag.tools.tool import Tool
+from neo4j_graphrag.types import LLMMessage
+
+
+class ToolsRetriever(Retriever):
+    """A retriever that uses an LLM to select appropriate tools for retrieval based on user input.
+
+    This retriever takes an LLM instance and a list of Tool objects as input. When a search is performed,
+    it uses the LLM to analyze the query and determine which tools (if any) should be used to retrieve
+    the necessary data. It then executes the selected tools and returns the combined results.
+
+    Args:
+        driver (neo4j.Driver): Neo4j driver instance.
+        llm (LLMInterface): LLM instance used to select tools.
+        tools (Sequence[Tool]): List of tools available for selection.
+        neo4j_database (Optional[str], optional): Neo4j database name. Defaults to None.
+        system_instruction (Optional[str], optional): Custom system instruction for the LLM. Defaults to None.
+    """
+
+    # Disable Neo4j version verification since this retriever doesn't directly interact with Neo4j
+    VERIFY_NEO4J_VERSION = False
+
+    def __init__(
+        self,
+        driver: neo4j.Driver,
```

Comment:
Should we consider updating the `Retriever` interface if this once does not use the driver?